lib/tasks/scripts/message_patients.rb

```diff
@@ -0,0 +1,85 @@
+class MessagePatients
+  def self.call(*args)
+    new(*args).call
+  end
+
+  attr_reader :patients, :message, :whatsapp, :dryrun, :verbose, :report
+
+  def initialize(patients, message, whatsapp: true, dryrun: false, verbose: true)
+    @patients = patients
+    @message = message
+    @whatsapp = whatsapp
+    @dryrun = dryrun
+    @verbose = verbose
+    @report = {}
+  end
+
+  def call
+    unless patients.is_a?(ActiveRecord::Relation)
+      raise ArgumentError, "Patients should be passed in as an ActiveRecord::Relation."
+    end
+
+    log("Total patients that will be contacted via #{whatsapp ? "whatsapp" : "sms"}: #{valid_patients.count}.")
+
+    return if dryrun
+
+    send_messages!
+    print_report
+    self
+  end
+
+  private
+
+  def send_messages!
```

Comment:
The `! = mutation/danger` convention is nice when you've got both versions of a method. In this case, I think we can drop the bangs, the methods are pretty self-explanatory. What do you think?